components/maze_solver/maze.cpp

```diff
@@ -0,0 +1,6 @@
+#include "maze_solver/maze.h"
+
+namespace micromouse
+{
+template class Maze<>;
```

Comment:
What is this file's purpose?

Reply:
It instantiates the template, necessary because the default template is marked `extern`.

It's supposed to reduce code size when using a template in many compilation units.